pkg/telemetry/counter.go

```diff
@@ -13,6 +13,10 @@ import (
 
 // Counter tracks how many times something is happening.
 type Counter interface {
+	// Initialize creates the counter with the given tags and initializes it to 0.
+	// This method is intended to be used when the counter value is important to
+	// send even before any incrementing/addition is done on it.
+	Initialize(tagsValue ...string)
```

Comment:
What's the behavior of `Counter.Initialize` when the counter with the given tags already exists?

Reading the implementation, I think I know the answer (it's a no-op), which is important for its usage in `check.NewStats` since check instances with the same name (but different IDs) can be initialized and in that case share the same (counter, tag values) couple which shouldn't be reset to `0` at each check instance's initialization. 

So I think this behavior should be documented and tested (not necessary for 7.26 though). WDYT @sgnn7 ?

Reply:
Your hunch is right - it's a noop afaict from the code. The code eventually ends in  Prometheus' [`getOrCreateMetricWithLabelValues()`](https://github.com/prometheus/client_golang/blob/b8fcd28885584356c562e9ecad42cbec45f4ce96/prometheus/vec.go#L384-L388) method for metrics:

```
// getOrCreateMetricWithLabelValues retrieves the metric by hash and label value
// or creates it and returns the new one.
```

I do agree when looking at your perspective of usage that the documentation could be improved a touch to indicate that this does not reset the value if the counter already exists so I'll keep that in mind and open a PR tp update it when we unfreeze master.